### ISSUE TITLE
[Proposal] Default locale and lang in Voilinux

### ISSUE THREAD

**Issue by yopito** [voidlinux/void-packages#9132]
### System

* xuname:   `Void 4.13.12_1 x86_64 GenuineIntel uptodate rF`
* package:  `glibc-locales-2.26_3`

### Expected behavior

following issue #8406 : `locale` should work of of the box on glibc system

### Actual behavior

On stock system, `locale` warns about missing resources:
```
$locale
locale: Cannot set LC_CTYPE to default locale: No such file or directory
locale: Cannot set LC_MESSAGES to default locale: No such file or directory
locale: Cannot set LC_ALL to default locale: No such file or directory
LANG=en_US.UTF-8
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_COLLATE=C
LC_MONETARY="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_PAPER="en_US.UTF-8"
LC_NAME="en_US.UTF-8"
LC_ADDRESS="en_US.UTF-8"
LC_TELEPHONE="en_US.UTF-8"
LC_MEASUREMENT="en_US.UTF-8"
LC_IDENTIFICATION="en_US.UTF-8"
LC_ALL=
```
default configuration is:
```
$ cat /etc/locale.conf 
LANG=en_US.UTF-8
LC_COLLATE=C
```

### Steps to reproduce the behavior

Install a system without any locale customisation

### Proposal

* change content of `/etc/default/libc-locales` : include "`en_US.UTF-8`" by default (`glibc-locales` package)
* then update `common/xbps-src/shutils/chroot.sh` : unedeed step if previous point is done.
* check comportment of void-installer about it.

If acknowledged, I'm ready to provide a PR for point 1 quickly. Let me know.

**Comment by Vaelatern** [comment]
Odd locale default if people are not in the US or speaking English. Any default would be arbitrary.

**Comment by yopito** [comment]
yes, but default settings have to be consistent, at least usable.

For instance, without "`en_US.UTF-8`" in libc-locales, some application don't even start:
```
$ gsmartcontrol
	Using the fallback 'C' locale.
<warn>  [hz] Warning: exit: Device open failed, or device did not return an IDENTIFY DEVICE structure.
<warn>  [app] SmartctlParser::parse_full(): No ATA sections could be parsed. Returning.
<warn>  [hz] Warning: exit: Device open failed, or device did not return an IDENTIFY DEVICE structure.
<warn>  [app] SmartctlParser::parse_section_info(): Unknown Info line encountered.
<warn>  [hz] FS warning: Error while opening file "/proc/scsi/sg/devices": No such file or directory.
<warn>  [app] {anonymous}::read_proc_scsi_sg_devices_file(): Sg devices file doesn't exist.
[yopito@oli73 ~]$ /usr/bin/gsmartcontrol-root

(process:7082): Gtk-WARNING **: Locale not supported by C library.
	Using the fallback 'C' locale.
<warn>  [hz] Warning: exit: Device open failed, or device did not return an IDENTIFY DEVICE structure.
<warn>  [app] SmartctlParser::parse_section_info(): Unknown Info line encountered.
<warn>  [hz] FS warning: Error while opening file "/proc/scsi/sg/devices": No such file or directory.
<warn>  [app] {anonymous}::read_proc_scsi_sg_devices_file(): Sg devices file doesn't exist.
<error> [gtk] 
<error> [gtk] unhandled exception (type std::exception) in signal handler:
<error> [gtk] what: locale::facet::_S_create_c_locale name not valid
<error> [gtk] 
```

**Comment by tornaria** [comment]
Could we have C.UTF-8 as default, using the [C locale definition from debian](https://anonscm.debian.org/cgit/pkg-glibc/glibc.git/tree/debian/patches/localedata/locale-C.diff)?

See also:
* https://sourceware.org/bugzilla/show_bug.cgi?id=17318
* https://sourceware.org/bugzilla/show_bug.cgi?id=21302
* https://bugzilla.redhat.com/show_bug.cgi?id=1361965

**Comment by yopito** [comment]
Sorry, I'm not clever enough to fully catch the implications of using "`LC_LANG=C.UTF-8`", whether as default or not. 
First wait for its inclusion in glibc upstream ?  
Have to check with musl libc too. 
@Vaelatern: some thoughts about it ?

**Comment by tornaria** [comment]
I don't have experience with musl, but it seems the [default locale](https://wiki.musl-libc.org/functional-differences-from-glibc.html#Default-locale)  is "C.UTF-8",
which is different from "C" as explained [here](https://wiki.musl-libc.org/functional-differences-from-glibc.html#The-C/POSIX-locale).

If I understand it correctly, the idea of "C.UTF-8" is to allow utf8 while not really making a choice of locale. All the other aspects of localization would work as in the default "C" locale, as compared to using "en_US.UTF-8" where there is localization into US english.

Some differences: default paper size, representation of numbers, times, etc, sort order, etc.

I guess "C.UTF-8" should be equivalent way is to unset LANG, unset LC_COLLATE, and only set LC_CTYPE to some UTF-8 locale. But this would still need some locale with ctypes for UTF-8 to be compiled.
